rerun_cpp/src/rerun/partitioned_component_batch.cpp

```diff
@@ -0,0 +1,54 @@
+#include "partitioned_component_batch.hpp"
+
+#include "arrow_utils.hpp"
+#include "c/rerun.h"
+
+#include <arrow/array/array_nested.h>
+#include <arrow/buffer.h>
+#include <arrow/c/bridge.h>
+
+namespace rerun {
+    std::shared_ptr<arrow::DataType> PartitionedComponentBatch::list_array_type_for(
+        std::shared_ptr<arrow::DataType> inner_type
+    ) {
+        return arrow::list(inner_type);
+    }
+
+    Result<PartitionedComponentBatch> PartitionedComponentBatch::from_batch_and_lengths(
+        ComponentBatch batch, const Collection<uint32_t>& lengths,
+        std::shared_ptr<arrow::DataType> list_array_type
+    ) {
+        // Convert lengths into offsets.
+        // TODO(andreas): Should we expose a version with offsets directly?
+        std::vector<uint32_t> offsets;
+        offsets.resize(lengths.size() + 1);
+        offsets[0] = 0;
+        for (size_t i = 0; i < lengths.size(); i++) {
+            offsets[i + 1] = offsets[i] + lengths[i];
+        }
```

Comment:
We should do some checking that the `batch.len() == offsets.back()`

Reply:
we do so aplenty on the rust side

Reply:
... but it's so cheap, to do right away, so why not.

Reply:
hmm let's not do it here. We might reasonably well decide that not the entire array needs to be consumed and I'd rather leave that decision centralized